ddtrace/contrib/sanic/patch.py

```diff
@@ -0,0 +1,115 @@
+import ddtrace
+from ddtrace import config
+from ddtrace.constants import ANALYTICS_SAMPLE_RATE_KEY
+from ddtrace.ext import SpanTypes, http
+from ddtrace.http import store_request_headers, store_response_headers
+from ddtrace.propagation.http import HTTPPropagator
+from ddtrace.utils.wrappers import unwrap as _u
+from ddtrace.vendor.wrapt import wrap_function_wrapper as _w
+
+import sanic
+
+from ...internal.logger import get_logger
+
+log = get_logger(__name__)
+
+config._add("sanic", dict(service=config._get_service(default="sanic"), distributed_tracing=True))
+
+
+def _extract_tags_from_request(request):
+    tags = {}
+
+    http_method = request.method
+    if http_method:
+        tags[http.METHOD] = http_method
+
+    http_url = request.url
+    if http_url:
+        tags[http.URL] = http_url
+
+    query_string = None
+    if config.sanic.trace_query_string:
+        query_string = request.query_string
+        if isinstance(query_string, bytes):
+            query_string = query_string.decode()
+        tags[http.QUERY_STRING] = query_string
+
+    return tags
+
+
+def _update_span_from_response(span, response):
+    span.set_tag(http.STATUS_CODE, response.status)
+    if 500 <= response.status < 600:
+        span.error = 1
+    store_response_headers(response.headers, span, config.sanic)
+    
+
+def patch():
+    """Patch the instrumented methods.
+    """
+    if getattr(sanic, "__datadog_patch", False):
+        return
+    setattr(sanic, "__datadog_patch", True)
+    _w("sanic", "Sanic.handle_request", patch_handle_request)
+
+
+def unpatch():
+    """Unpatch the instrumented methods.
+    """
+    _u(sanic.Sanic, "handle_request")
+    if not getattr(sanic, '__datadog_patch', False):
+        return
+    setattr(sanic, '__datadog_patch', False)
+
+
+def patch_handle_request(wrapped, instance, args, kwargs):
+    """Wrapper for Sanic.handle_request"""
+
+    request, write_callback, stream_callback = args
+
+    resource = "{} {}".format(request.method, request.path)
+
+    headers = {k: v for (k, v) in request.headers.items()}
+
+    if config.sanic.distributed_tracing:
+        propagator = HTTPPropagator()
+        context = propagator.extract(headers)
+        if context.trace_id:
+            ddtrace.tracer.context_provider.activate(context)
+
+    with ddtrace.tracer.trace(
```

Comment:
Similar to the [comment I made on the ASGI integration](https://github.com/DataDog/dd-trace-py/pull/1567#discussion_r458333341), I'm a bit concerned about context propagation here, especially in Python 3.6 in this case.